src/micro_args.c

```diff
@@ -0,0 +1,355 @@
+// Copyright 2023 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "microp.h"
+#include "micro_args.h"
+
+struct args_s
+{
+    void **args;
+    int count;
+};
+
+static microError *parse(void **args, int *args_len, const char *data, int data_len);
+
+static inline microError *new_args(microArgs **ptr, int n)
+{
+    *ptr = NATS_CALLOC(1, sizeof(microArgs));
+    if (*ptr == NULL)
+        return micro_ErrorOutOfMemory;
+
+    (*ptr)->args = NATS_CALLOC(n, sizeof(void *));
+    if ((*ptr)->args == NULL)
+    {
+        NATS_FREE(*ptr);
+        return micro_ErrorOutOfMemory;
+    }
+
+    (*ptr)->count = n;
+    return NULL;
+}
+
+microError *
+micro_ParseArgs(microArgs **ptr, const char *data, int data_len)
+{
+    microError *err = NULL;
+    microArgs *args = NULL;
+    int n;
```

Comment:
Got a warning here on a compiler about this not being initialized.

Reply:
Curious - I don't seem to see it anywhere. It was harmless, but I'd like to understand how I missed the warning, where did you see it?